packages/zendesk-adapter/src/filters/guide_order_utils.ts

```diff
@@ -0,0 +1,135 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// Lowest position index first, if there is a tie - the newer is first
+import {
+  Change, ElemID, InstanceElement,
+  isAdditionChange, isReferenceExpression,
+  isRemovalChange,
+  ModificationChange, ObjectType, ReadOnlyElementsSource,
+} from '@salto-io/adapter-api'
+import { detailedCompare } from '@salto-io/adapter-utils'
+import { collections } from '@salto-io/lowerdash'
+import { deployChange, deployChanges } from '../deployment'
+import ZendeskClient from '../client/client'
+import { FilterContext } from '../config'
+import { ARTICLE_TYPE_NAME, CATEGORY_TYPE_NAME, SECTION_TYPE_NAME, ZENDESK } from '../constants'
+
+const { awu } = collections.asynciterable
+
+export const CATEGORIES_FIELD = 'categories'
+export const SECTIONS_FIELD = 'sections'
+export const ARTICLES_FIELD = 'articles'
+
+const orderFieldToType : {[key: string]: ObjectType} = {
+  [CATEGORIES_FIELD]: new ObjectType({ elemID: new ElemID(ZENDESK, CATEGORY_TYPE_NAME) }),
+  [SECTIONS_FIELD]: new ObjectType({ elemID: new ElemID(ZENDESK, SECTION_TYPE_NAME) }),
+  [ARTICLES_FIELD]: new ObjectType({ elemID: new ElemID(ZENDESK, ARTICLE_TYPE_NAME) }),
+}
+
+/* Split the changes into 2 groups:
+  withOrderChanges    - Changes with order changes
+  onlyNonOrderChanges - Changes without any order changes
+ */
+export const sortChanges = (changes: Change<InstanceElement>[], orderField: string) :
+    {
+      withOrderChanges : ModificationChange<InstanceElement>[]
+      onlyNonOrderChanges : Change<InstanceElement>[]
+    } => {
```

Comment:
styling nits
1. not sure why we don't enforce it in the linter, but we don't have spaces before `:`
3. this can also be
```suggestion
export const sortChanges = (changes: Change<InstanceElement>[], orderField: string): {
  withOrderChanges: ModificationChange<InstanceElement>[]
  onlyNonOrderChanges: Change<InstanceElement>[]
} => {
```
(up to you if you prefer this style)